lib/rules/template-tag-spacing.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @fileoverview Rule to check spacing between template tags and their literals
+ * @author Jonathan Wilsson
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "require or disallow spacing between template tags and their literals",
+            category: "ECMAScript 6",
+            recommended: false
+        },
+
+        fixable: "whitespace",
+
+        schema: [
+            { enum: ["always", "never"] }
+        ]
+    },
+
+    create(context) {
+        const never = context.options[0] !== "always";
+        const sourceCode = context.getSourceCode();
+
+        /**
+         * Check if a space is present between a template tag and its literal
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function checkSpacing(node) {
+            const tagToken = sourceCode.getLastToken(node.tag);
+            const literalToken = sourceCode.getTokenAfter(tagToken);
```

Comment:
@not-an-aardvark Won't the first line in your example also spit out the `)` token? Should that be `sourceCode.getFirstToken(node.tag)`?